sktalk/corpus/parsing/cha.py

```diff
@@ -5,32 +5,67 @@
 
 
 class ChaFile(InputFile):
+    TIMING_REGEX = r"\D(?P<timing>\d{1,9}_\d{1,9})\D"
+    # participant may not be repeated on each line
+    PARTICIPANT_REGEX = r"(^\*(?P<participant>[^:*]+)\:){0,1}"
+    UTTERANCE_REGEX = r"\s+(?P<utterance>.*)\s+"
+    LINE_REGEX = PARTICIPANT_REGEX + UTTERANCE_REGEX + TIMING_REGEX
+
+    SPACER_REGEX = r"\((?P<spacer>[\d.]+)\)"
+
     def _pla_reader(self) -> pylangacq.Reader:
         return pylangacq.read_chat(self._path)
 
     def _extract_metadata(self):
         return self._pla_reader().headers()[0]
 
     def _extract_utterances(self):
-        reader = self._pla_reader().utterances(by_files=False)
-        return [self._to_utterance(read_utterance) for read_utterance in reader]
-
-    @classmethod
-    def _to_utterance(cls, read_utterance) -> Utterance:
-        """Convert pylangacq Utterance to sktalk Utterance"""
-        participant = read_utterance.participant
-        time = read_utterance.time_marks
-        time = list(time) if isinstance(time, (list, tuple)) else None
-        text = str(read_utterance.tiers)
-        text = cls._clean_utterance(text)
-        return Utterance(
-            participant=participant,
-            time=time,
-            utterance=text,
-        )
-
-    @classmethod
-    def _clean_utterance(cls, utterance):
+        with open(self._path, encoding="utf-8") as f:
+            lines = f.readlines()
+        utterance_info = [self._extract_info(
+            line) for line in lines if not line.startswith("@")]
+
+        # collect all utterance info in a terrible, terrible loop
+        collection = []
+        participant = None
+        for info in utterance_info:
+            if info["utterance"] is None:
+                continue
+            if info["participant"] is not None:
+                participant = info["participant"]
+            complete_utterance = Utterance(
+                participant=participant,
+                time=info["time"],
+                utterance=info["utterance"])
+            collection.append(complete_utterance)
+        return collection
+
+    @staticmethod
+    def _extract_info(line):
+        default_return = {"utterance": None}
+
+        extract_re = re.search(ChaFile.LINE_REGEX, line)
+        if not bool(extract_re):
```

Comment:
I think that `extract_re` is `None` if there is no match, so that you could do:
```suggestion
        if not extract_re:
```
https://docs.python.org/3/library/re.html#match-objects